src/vdaf.rs

```diff
@@ -210,7 +210,12 @@ pub trait Aggregator<const VERIFY_KEY_SIZE: usize, const NONCE_SIZE: usize>: Vda
     ///
     /// Decoding takes a [`Self::PrepareState`] as a parameter; this [`Self::PrepareState`] may be
     /// associated with any aggregator involved in the execution of the VDAF.
-    type PrepareMessage: Clone + Debug + ParameterizedDecode<Self::PrepareState> + Encode;
+    type PrepareMessage: Clone
+        + Debug
+        + PartialEq
+        + Eq
```

Comment:
I added `ParialEq + Eq` here and on `type PrepareState` so that `topology::ping_pong::Transition` could in turn be `PartialEq + Eq`. The relevant types for `prio3` and `poplar1` are already `PartialEq + Eq` so I think this change is harmless, and may let us delete some trait bound noise from DAP impementations.

Reply:
Why is `ReportAggregationState` `PartialEq`?

If `PrepareState` needs to be `PartialEq`, then we need to make sure the implementation is constant-time. That way we make it less likely that we leak bits of the state via a timing channel.

Reply:
Ack. I still think that comparing state should be constant time, but we can just file an issue for now.

Reply:
https://github.com/divviup/libprio-rs/issues/722